astropy/time/core.py

```diff
@@ -199,18 +196,20 @@ def __init__(self, val, val2=None, format=None, scale=None,
         else:
             self.location = None
 
-        if precision is not None:
-            self.precision = precision
-        if in_subfmt is not None:
-            self.in_subfmt = in_subfmt
-        if out_subfmt is not None:
-            self.out_subfmt = out_subfmt
-
         if isinstance(val, self.__class__):
```

Comment:
I realize this was already here, but why `isinstance(val, self.__class__)` and not just `isinstance(val, Time)`?